PORTAL/jobs/requests.py

```diff
@@ -40,24 +40,27 @@ def from_raw(cls, raw: Any):
     @classmethod
     def parse(cls, idstr: str):
         kinds = '|'.join(cls._KIND_BY_VALUE)
-        m = re.match(rf'^req-(?:({kinds})-)?(\d{{10}})-(\w+)$', idstr)
+        m = re.match(rf'^req-(?:({kinds})-)?(\d{{10}})-(\w+)(?:-(\w+))?$', idstr)
```

Comment:
FWIW, at a certain point a pattern gets too complex for a single line.  I tend to switch to using `re.VERBOSE` mode in those cases.  For example see `Version.REGEX` in _utils.py.  (The pattern doesn't *have* to be a separate variable, but it often helps readability when using a regex flag.)

This case is borderline.  We could probably leave it as-is.